subprojects/dependency-management/src/main/java/org/gradle/internal/resolve/result/BuildableModuleComponentMetaDataResolveResult.java

```diff
@@ -70,4 +70,9 @@
     boolean isAuthoritative();
 
     void setAuthoritative(boolean authoritative);
+
+    /**
+     * Returns true if we should retry to get the resource when it was missing
+     */
+    boolean isRetryMissing();
```

Comment:
:x: Having the `BuildableModuleComponentMetaDataResolveResult.retryMissing` flag isn't required: instead, we can just interpret the result differently based in the caller:
- The result will return 'authoritative' or not based on the facts, not based on any flags from the caller
- The caller `ComponentMetadataResolveState` can hold `retryMissing` in a field, and decide whether to set the "searchedRemotely" flag based on a) `result.authoritative` b) `result.state == MISSING` and c) `retryMissing`

This retains the true value of the `authoritative` flag, and avoids having a control flag on the `BuildableModuleComponentMetaDataResolveResult`, which is really an output parameter, not an input to the resolution.